expression/builtin_arithmetic.go

```diff
@@ -173,66 +176,149 @@ func (c *arithmeticPlusFunctionClass) getFunction(ctx sessionctx.Context, args [
 		return sig, nil
 	} else {
 		bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETInt, types.ETInt, types.ETInt)
-		if mysql.HasUnsignedFlag(args[0].GetType().Flag) || mysql.HasUnsignedFlag(args[1].GetType().Flag) {
+		isLHSUnsigned := mysql.HasUnsignedFlag(args[0].GetType().Flag)
+		isRHSUnsigned := mysql.HasUnsignedFlag(args[1].GetType().Flag)
+		if isLHSUnsigned || isRHSUnsigned {
 			bf.tp.Flag |= mysql.UnsignedFlag
 		}
 		setFlenDecimal4Int(bf.tp, args[0].GetType(), args[1].GetType())
-		sig := &builtinArithmeticPlusIntSig{bf}
-		sig.setPbCode(tipb.ScalarFuncSig_PlusInt)
+		var sig builtinFunc
+		switch {
+		case isLHSUnsigned && isRHSUnsigned:
+			sig = &builtinArithmeticPlusIntUnsignedUnsignedSig{bf}
+			sig.setPbCode(tipb.ScalarFuncSig_PlusIntUnsignedUnsigned)
+		case isLHSUnsigned && !isRHSUnsigned:
+			sig = &builtinArithmeticPlusIntUnsignedSignedSig{bf}
+			sig.setPbCode(tipb.ScalarFuncSig_PlusIntUnsignedSigned)
+		case !isLHSUnsigned && isRHSUnsigned:
+			sig = &builtinArithmeticPlusIntSignedUnsignedSig{bf}
+			sig.setPbCode(tipb.ScalarFuncSig_PlusIntSignedUnsigned)
+		case !isLHSUnsigned && !isRHSUnsigned:
+			sig = &builtinArithmeticPlusIntSignedSignedSig{bf}
+			sig.setPbCode(tipb.ScalarFuncSig_PlusIntSignedSigned)
+		default:
```

Comment:
Will this default actually being executed?

Reply:
It's part of my test .I delete this.